components/pin-input.js

```diff
@@ -0,0 +1,144 @@
+class PinInput extends /** @type {any} */ Nimiq.Observable {
+    /**
+     * @param {Element|undefined} el
+     */
+    constructor(el = undefined) {
+        super();
+        this.$el = el || this._createElement();
+        window.$el = this.$el; // TODO remove
+        this.$dotIndicator = /** @type {Element} */ (this.$el.querySelector('.pin-dot-indicator'));
+        this.$dots = Array.from(this.$dotIndicator.children);
+        this.$deleteButton = /** @type {Element} */ (this.$el.querySelector('.delete'));
+        /** @type {string} */
+        this._pin = '';
+        /** @type {number} */
+        this._attempts = 0;
+        /** @type {number} */
+        this._waitingTime = 50;
+        /** @type {function(KeyboardEvent):void} */
+        this._handleKeyboardInput = this._handleKeyboardInput.bind(this);
+
+        this.$el.addEventListener('click', e => this._onClick(/** @type {MouseEvent} */ (e)));
+        this.$deleteButton.addEventListener('click', () => this._onDelete());
+    }
+
+    /** @returns {Element} */
+    _createElement() {
+        /** @type {Element} */
+        const el = document.createElement('div');
+        el.classList.add('pin-input');
+        el.classList.add('center');
```

Comment:
could be "el.classList.add('pin-input', 'center');"

Reply:
didn't know about that syntax. thanks